src/components/sidebar/NoteSidebar.tsx

```
@@ -108,162 +109,184 @@ export function NoteSidebar() {
     e.preventDefault();
     if (!editingItemId) return;
 
-    const folder = folders.find((f) => f.id === editingItemId);
-    if (folder) {
-      handleUpdateFolder(editingItemId, { name: editingName });
+  // Memoized filtered notes based on active folder
+  const filteredNotes = useMemo(
```

ESLint

[error] React Hook "useMemo" is called in function "handleRenameSubmit" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".

(react-hooks/rules-of-hooks)